File: c/sorting_algorithms/Bubble Sort/bubbleSort.c
#include <stdio.h>

void bubbleSort(int elements[],int size){
    // Loop to access each element in an array.
    for(int step = 0; step < size - 1; ++step){
        // Loop to compare each element in an array.
        for(int i = 0; i < size - step - 1; ++i){
            if(elements[i] > elements[i + 1]){
                int temp = elements[i];
                elements[i] = elements[i + 1];
                elements[i + 1] = temp;
            };
        };
    };
};

int main(){
    int elements[] = {5,18,0,-3,28,-2};

    int size = sizeof(elements) / sizeof(elements[0]);

    // Print Original Array
    printf("Original Array: \n");
     for (int i = 0; i < size; ++i) {
        printf("%d  ", elements[i]);
    }   
    printf("\n");

    bubbleSort(elements, size); 

    // Print Sorted Array 
    printf("Sorted Elements In An Ascending Order: \n");
 
    for (int i = 0; i < size; ++i) {
        printf("%d  ", elements[i]);
    }   
    printf("\n");
};

File: python/sorting_algorithms/bubble_sort/bubbleSort.py
def bubbleSort(array):
    # loop through each element of array
    for step in range(len(array)):
        swapped = False

        # loop to compare array elements
        for j in range(0,len(array) - step -1):
            # compare two adjacent elements
            # change > to < to sort in descending order.
            temporary = array[j]
            array[j] = array[j + 1]
            array [j + 1] = temporary

            swapped = True
        
        if not swapped:
            break

element = [54,88,32,6,-2]

print('UnSorted Array:')
print(element)

bubbleSort(element)

print('Sorted Array in Ascending Order:')
print(element)